Common/cpp/Tools/SingleInstanceChecker.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+namespace reanimated {
+
+using namespace std;
+
+// This is a class that counts how many instances of a different class there
+// are. It is meant only to be used with classes that should only have one
+// instance.
+
+template <class T>
+class SingleInstanceCheker {
+ public:
+  SingleInstanceCheker();
+  ~SingleInstanceCheker();
+
+ private:
+  void assertWithMessage(bool condition, string message) {
+    if (!condition) {
+      cerr << message << endl;
+      assert(condition);
+    }
+  }
+
+  // A static field will exist separatley for every class template.
+  inline static int instanceCount;
```

Comment:
1. Why is `inline` necessary here?
2. Rename to `instanceCount_`

Reply:
Inline is necessary for the field to be automatically initialized. I'll add a comment